cluster/src/main/scala/org/apache/spark/ui/SnappyMemberDetailsPage.scala

```diff
@@ -0,0 +1,463 @@
+        Showing {curLogLength} Bytes: {startByte.toString} - {endByte.toString} of {logLength}
+      </span>
+
+    val moreButton =
+      <button type="button" onclick={"loadMore()"} class="log-more-btn btn btn-default">
+        Load More
+      </button>
+
+    val newButton =
+      <button type="button" onclick={"loadNew()"} class="log-new-btn btn btn-default">
+        Load New
+      </button>
+
+    val alert =
+      <div class="no-new-alert alert alert-info" style="display: none;">
+        End of Log
+      </div>
+
+    val logParams = "/?memId=%s".format(memberId)
+
+    val jsOnload = "window.onload = " +
+        s"initLogPage('$logParams', $curLogLength, $startByte, $endByte, $logLength, $byteLength);"
+
+    val content =
+      <div style="margin-top:5px; margin-left:15px;">
+        {range}
+        <div class="log-content"
+             style="height:60vh; overflow:auto; margin-top:5px; border: 1px solid #E2E2E2;">
+          <div>{moreButton}</div>
+          <pre>{logText}</pre>
+          {alert}
+          <div>{newButton}</div>
+        </div>
+        <script>{Unparsed(jsOnload)}</script>
+      </div>
+
+    PageContent = pageTitleNode ++ memberStats ++ memberLogTitle ++ content
+
+    UIUtils.simpleSparkPageWithTabs_2("Member Details Page", PageContent, parent, Some(500))
+  }
+
+  def renderLog(request: HttpServletRequest): String = {
+
+    val memberId = Option(request.getParameter("memId")).map { memberId =>
+      UIUtils.decodeURLParameter(memberId)
+    }.getOrElse {
+      throw new IllegalArgumentException(s"Missing memId parameter")
+    }
+
+    val offset = Option(request.getParameter("offset")).map(_.toLong)
+    val byteLength = Option(request.getParameter("byteLength")).map(_.toInt).getOrElse(defaultBytes)
+
+    val (logText, startByte, endByte, logLength) =
+      getLog(workDir.getPath, logFileName, offset, byteLength)
+
+    val pre = s"==== Bytes $startByte-$endByte of $logLength of ${workDir.getPath}/$logFileName ====\n"
+
+    pre + logText
+
+  }
+
+  /** Get the part of the log files given the offset and desired length of bytes */
+  private def getLog(
+      logDirectory: String,
+      logFile: String,
+      offsetOption: Option[Long],
+      byteLength: Int
+      ): (String, Long, Long, Long) = {
+
+    if (logFile == null || logFile.isEmpty) {
+      return ("Error: Log file must be specified ", 0, 0, 0)
+    }
+
+    // Verify that the normalized path of the log directory is in the working directory
+    val normalizedUri = new File(logDirectory).toURI.normalize()
+    val normalizedLogDir = new File(normalizedUri.getPath)
+    if (!Utils.isInDirectory(workDir, normalizedLogDir)) {
+      return ("Error: invalid log directory " + logDirectory, 0, 0, 0)
+    }
+
+    try {
+      val files = RollingFileAppender.getSortedRolledOverFiles(logDirectory, logFile)
+      logDebug(s"Sorted log files of type $logFile in $logDirectory:\n${files.mkString("\n")}")
```

Comment:
Thanks Sumedh, as discussed I will make changes to load logs from local file system on member machine.